src/Microsoft.Diagnostics.Runtime/src/Common/HeapBase.cs

```diff
@@ -263,24 +275,37 @@ protected internal override IEnumerable<ClrObjectReference> EnumerateObjectRefer
             else
                 Debug.Assert(type == GetObjectType(obj));
 
-            if (type == null || !type.ContainsPointers)
+            if (type == null || (!type.ContainsPointers && !type.IsCollectible))
```

Comment:
Do collectible types not have references?  I'm confused as to why this check is here but I'm not very familiar with the feature.

Reply:
There is no difference in `ContainsPointers`: it still returns `false` for unmanaged types.

Reply:
> Do collectible types not have references?

Its negated, collectible types can have references (the loader allocator) even if the type doesn't contain pointers